supervisor/docker/addon.py

```diff
@@ -350,6 +352,38 @@ def mounts(self) -> list[Mount]:
                 )
             )
 
+        # Map addon's public config folder if not using deprecated config option
+        else:
```

Comment:
I guess the idea was to use a config option for enable that feature, so you see only addon folders inside configs which also use that feature?

Reply:
As discussed, I agree we should have a configuration to enable the per add-on config, so that add-ons which don't use it don't get one (avoids unnecessary directories, which comes especially into play for add-ons which will show all this add-on config folders such as VS Code add-on).